pulse2percept/models/granley2021.py

```diff
@@ -582,14 +586,18 @@ def predict_percept(self, implant, t_percept=None):
         if not isinstance(implant.stim, BiphasicPulseTrain):
             # Could still be a stimulus where each electrode has a biphasic pulse train
             # or a 0 stimulus
-            for i, (ele, params) in enumerate(implant.stim.metadata
-                                              ['electrodes'].items()):
-                if (params['type'] != BiphasicPulseTrain or
-                        params['metadata']['delay_dur'] != 0) and \
-                        np.any(implant.stim[i]):
-                    raise TypeError(
-                        f"All stimuli must be BiphasicPulseTrains with no " +
-                        f"delay dur (Failing electrode: {ele})")
+            try:
+                for i, (ele, params) in enumerate(implant.stim.metadata
+                                                ['electrodes'].items()):
+                    if (params['type'] != BiphasicPulseTrain or
+                            params['metadata']['delay_dur'] != 0) and \
+                            np.any(implant.stim[i]):
+                        raise TypeError(
+                            f"All stimuli must be BiphasicPulseTrains with no " +
+                            f"delay dur (Failing electrode: {ele})")
+            except KeyError:
+                raise KeyError(f"All stimuli must be BiphasicPulseTrains with no " +
```

Comment:
I think this should be a type error: The problem is that the passed stimuli was not the correct type